src/test/rpc/GetCounts_test.cpp

```diff
@@ -0,0 +1,119 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/SField.h>
+
+namespace ripple {
+
+class GetCounts_test : public beast::unit_test::suite
+{
+    void testGetCounts()
+    {
+        using namespace test::jtx;
+
+        Env env(*this);
+        // check counts with no transactions posted
+        auto jrr = env.rpc("get_counts")[jss::result];
+        BEAST_EXPECT(jrr[jss::status] == "success");
+        BEAST_EXPECT(! jrr.isMember("Transaction"));
+        BEAST_EXPECT(! jrr.isMember("STObject"));
+        BEAST_EXPECT(! jrr.isMember("HashRouterEntry"));
```

Comment:
I looked for counted objects in the subsequent request (after there are some actual counts reported) and then just verified that those same fields are not present in this initial request. This is definitely not exhaustive and probably could be improved.